CHANGELOG.md

```diff
@@ -123,6 +123,10 @@ Adding a new version? You'll need three changes:
   now accounts for potential failures in synchronizing configuration with Konnect's
   Runtime Group Admin API.
   [#4029](https://github.com/Kong/kubernetes-ingress-controller/pull/4029)
+- Record an event attached to KIC pod after applying configuration to Kong. If
+  the applying succeeded, a `Normal` event is recorded. If the applying failed,
+  a `Warning` event is recorded.
+  [#4054](https://github.com/Kong/kubernetes-ingress-controller/pull/4054)
```

Comment:
```suggestion
- Record an event attached to KIC pod after applying configuration to Kong. If
  the applying succeeded, a `Normal` event with `KongConfigurationSucceeded` 
  reason is recorded. If the applying failed, a `Warning` event with 
  `KongConfigurationApplyFailed` reason is recorded.
  [#4054](https://github.com/Kong/kubernetes-ingress-controller/pull/4054)
```

I think that might be useful to tell users what reasons they should expect so they can filter events using those.